lib/scsi_defs.h

```diff
@@ -97,5 +99,77 @@ namespace scsi_defs {
   } ABSL_ATTRIBUTE_PACKED;
   static_assert(sizeof(InquiryCommand) == kInquiryCommandSize);
 
+
+  // Refer to https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf , Section 3.6.2 Table 60
+  enum class PeripheralQualifier: uint8_t {
+    kPeripheralDeviceConnected = 0b000,
+    kPeripheralDeviceNotConnected = 0b001,
+    kReserved = 0b010,
+    kNotSupported = 0b011
+  };
+
+  // Refer to https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf , Section 3.6.2 Table 63
+  enum class TPGS : uint8_t {
+    kNotSupported = 0b00,
+    kImplicitAccess = 0b01,
+    kExcplicitAccess = 0b10,
+    kFullAccess = 0b11,
+  };
+
+  enum class ResponseDataFormat : uint8_t {
+    kObsolete0 = 0x0,
+    kObsolete1 = 0x1,
+    kCompliant = 0x2,
+  };
+
+  // Refer to https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf , Section 3.6.2 Table 59
+  // Fields are arranged according to Big Endian format
+  struct InquiryData {
+    PeripheralQualifier peripheral_qualifier: 3;
+    PeripheralDeviceType peripheral_device_type: 5;
+    bool rmb: 1; // Removable Media Bit (RMB)
+    bool lu_cong: 1;
+    uint8_t reserved_1: 6;
+    Version version: 8;
+    bool reserved_2: 2;
+    bool normaca: 1; //Normal ACA (NORMACA)
+    bool hisup: 1; // Hierarchical Support Bit (HISUP)
+    ResponseDataFormat response_data_format: 4;
+    uint8_t additional_length: 8;
+    bool sccs: 1; // SCC Supported
+    bool acc: 1; // Access controls Coordinator Bit
+    TPGS tpgs: 2; // Target Port Group Support (TPGS)
+    bool third_party_copy: 1; // referred to as 3PC in the documentation
+    uint8_t reserved_3: 2;
+    bool protect: 1;
+    bool obsolete_1: 1;
+    bool encserv: 1; // Enclosure Services Bit
+    bool vs_1: 1; // vendor specific bit
+    bool multip: 1; // multiple SCSI Port
+    bool obsolete_2: 1;
+    bool reserved_4: 2;
+    bool addr_16: 1; // SCSI 16-bit address support bit
+    bool obsolete_3: 1;
+    bool reserved_5: 1;
+    bool wbus_16: 1; // Wide Bus bit 
+    bool sync: 1;
+    bool obsolete_4: 1;
+    bool reserved_6: 1;
+    bool cmdque: 1; // Command Management Model bit
+    bool vs_2: 1; // vendor specific bit
+    uint8_t vendor_identification[8];
+    uint8_t product_identification[16];
+    uint8_t product_revision_level[4];
+    uint8_t vendor_specific_1[20];
+    uint8_t reserved_7: 4;
+    uint8_t clocking: 2;
+    bool qas: 1; // Quick Arbitration and Selection Supported bit
+    bool ius: 1; // Information Units Supported bit
+    uint8_t reserved_8: 8;
+    uint16_t vendor_descriptor[8];
+    uint8_t reserved_9[22];
+  } ABSL_ATTRIBUTE_PACKED; 
+  static_assert(sizeof(InquiryData) >= kInquiryDataSize);
```

Comment:
should this be `>= 96`.

Reply:
Why are we asserting greater than or equal to and not just equal to?

Reply:
i think the `>=` is because of the variable size fields

Reply:
@eyhhuang the official documentation says that this structure shall contain at least 36 bytes and that's why I am asserting for that value.

@harloff-g This structure had a variable sized field and the size of this structure is at least 36 bytes. That's why I was checking for `>= 36` but since we're not including the variable sized fields as of now, I change it to assert for the exact value.

Reply:
I think we will need to check for 96 bytes in the static assert but will need to have a check in the parsing logic later on for ensuring at least 36 bytes are passed to this structure.

Reply:
Yeah that's kind of what I was thinking as well. The struct will always be 96 bytes but some fields might not be valid depending on the value of the additional_length field

Reply:
Yup so I updated the assert to check for 96 here. We can enforce the 36 byte requirement later on